### PR TITLE
Disable delete button

### PR DESCRIPTION
**Pull request by Mallusca**

### CODE REVIEW

File: SchoolJournal.Web/Controllers/AdminGroupsController.cs [hunk 1]
```diff
@@ -26,7 +27,7 @@ public ActionResult GroupsPage()
         {
             AdminGroupsPageViewModel pageModel = new AdminGroupsPageViewModel
             {
-                Groups = _groupsService.List()
+                Groups = _groupsService.List(),
```
Comment (AJIex1s) [inline SchoolJournal.Web/Controllers/AdminGroupsController.cs:30]: запятая!

File: SchoolJournal.Web/Views/Admin/Groups/GroupsGrid.cshtml [hunk 2]
```diff
@@ -185,7 +186,7 @@
             <td class="edit-group-selection">
                 <div class="d-flex justify-content-between">
                     <i onclick="editGroup(@group.Id)" class="fa fa-edit icon-hover edit-group  icon-wrapped" title="Редактировать группу!"></i>
-                    <i onclick="deleteGroup(@group.Id)" class="fa fa-trash-o icon-wrapped icon-hover delete-group-@group.Id"></i>
+                    <i onclick="deleteGroup(@group.Id)" title ="@(group.StudentInGroup.ToString()=="False"?"":"Вы не можете удалить группу со студентами!")"class="fa fa-trash-o icon-wrapped @(group.StudentInGroup.ToString()=="False"?"icon-hover":"disabled" )"></i>
```
Comment (AJIex1s) [inline SchoolJournal.Web/Views/Admin/Groups/GroupsGrid.cshtml:189]: https://stackoverflow.com/questions/8061647/conditional-html-attributes-using-razor-mvc3

Comment (AJIex1s) [reply SchoolJournal.Web/Views/Admin/Groups/GroupsGrid.cshtml:189]: attribute title can be simplified

Comment (AJIex1s) [reply SchoolJournal.Web/Views/Admin/Groups/GroupsGrid.cshtml:189]: Why you are converting StudentInGroup to string? You can avoid it.

Comment (AJIex1s) [reply SchoolJournal.Web/Views/Admin/Groups/GroupsGrid.cshtml:189]: Terrible formatting here. Spaces.

Comment (AJIex1s) [reply SchoolJournal.Web/Views/Admin/Groups/GroupsGrid.cshtml:189]: You can assign icon-hover class unconditionally - .icon-hover.disabled rule must be added to stylesheet.

File: SchoolJournal.Web/wwwroot/css/groups.css [hunk 3]
```diff
@@ -65,3 +65,12 @@ textarea {
     height: 32px;
     width: 130px;
 }
+
+.disabled,
+.disabled:hover {
+    color: grey;
+    content: attr(title);
+}
+    .disabled:active {
```
Comment (AJIex1s) [inline SchoolJournal.Web/wwwroot/css/groups.css:74]: why you added active? 
you can assign rule - pointer-events: none; to .disabled

Comment (AJIex1s) [reply SchoolJournal.Web/wwwroot/css/groups.css:74]: content: attr(title); for what purpose?

Comment (AJIex1s) [reply SchoolJournal.Web/wwwroot/css/groups.css:74]: color: grey - you can find more suitable color..

File: SchoolJournalCore.DAL.Interfaces/IGroupsRepository.cs [hunk 4]
```diff
@@ -13,7 +13,7 @@ public interface IGroupsRepository
     {
         IQueryable<GroupDomainModel> GetGroups();
         Task<long> CreateGroup(CreateGroupDomainModel model);
-        public IQueryable<string> GroupNames();
+        //public IQueryable<string> GroupNames();
```
Comment (AJIex1s) [inline SchoolJournalCore.DAL.Interfaces/IGroupsRepository.cs:16]: comment

File: SchoolJournalCore.DAL/Groups/GroupsRepository.cs [hunk 5]
```diff
@@ -30,7 +30,6 @@ public IQueryable<GroupDomainModel> GetGroups()
             //var data = (from gr in DbContext.Groups
             //            join t in DbContext.Teachers on gr.TeacherId equals t.Id,
             //            );
```
Comment (AJIex1s) [inline SchoolJournalCore.DAL/Groups/GroupsRepository.cs:32]: comment

File: SchoolJournalCore.DAL/Groups/GroupsRepository.cs [hunk 6]
```diff
@@ -41,7 +40,8 @@ public IQueryable<GroupDomainModel> GetGroups()
                     Description = gr.Description,
                     WorkingHours = gr.WorkingHours.Select(x => x.Adapt<WorkingHoursDomainModel>()),
                     TeacherId = gr.TeacherId,
-                    TeacherName = gr.Teacher != null ? gr.Teacher.FirstName + gr.Teacher.LastName : null
+                    TeacherName = gr.Teacher != null ? gr.Teacher.FirstName + gr.Teacher.LastName : null,
+                    StudentInGroup = DbContext.Students.Any(x=>x.GroupId == gr.Id)
```
Comment (AJIex1s) [inline SchoolJournalCore.DAL/Groups/GroupsRepository.cs:44]: formatting

Comment (AJIex1s) [reply SchoolJournalCore.DAL/Groups/GroupsRepository.cs:44]: You must use join instead of subrequest

File: SchoolJournalCore.Domain/GroupDomainModel.cs [hunk 7]
```diff
@@ -17,6 +17,8 @@ public class GroupDomainModel
 
         public string Description { get; set; }
 
+        public bool StudentInGroup { get; set; }
```
Comment (AJIex1s) [inline SchoolJournalCore.Domain/GroupDomainModel.cs:20]: rename - ContainsStudents\HasStudents

File: SchoolJournalCore.ViewModels/Groups/GroupViewModel.cs [hunk 8]
```diff
@@ -17,6 +17,8 @@ public class GroupViewModel
 
         public string Description { get; set; }
 
+        public bool StudentInGroup { get; set; }
```
Comment (AJIex1s) [inline SchoolJournalCore.ViewModels/Groups/GroupViewModel.cs:20]: rename

File: SchoolJournal.Web/Controllers/AdminGroupsController.cs (hunk 1 above)
Comment (Mallusca) [reply SchoolJournal.Web/Controllers/AdminGroupsController.cs:30]: done

File: SchoolJournal.Web/Views/Admin/Groups/GroupsGrid.cshtml (hunk 2 above)
Comment (Mallusca) [reply SchoolJournal.Web/Views/Admin/Groups/GroupsGrid.cshtml:189]: done

File: SchoolJournal.Web/wwwroot/css/groups.css (hunk 3 above)
Comment (Mallusca) [reply SchoolJournal.Web/wwwroot/css/groups.css:74]: content: attr(title); for what purpose? output title

File: SchoolJournalCore.DAL.Interfaces/IGroupsRepository.cs (hunk 4 above)
Comment (Mallusca) [reply SchoolJournalCore.DAL.Interfaces/IGroupsRepository.cs:16]: done

File: SchoolJournalCore.DAL/Groups/GroupsRepository.cs (hunk 5 above)
Comment (Mallusca) [reply SchoolJournalCore.DAL/Groups/GroupsRepository.cs:32]: done

File: SchoolJournal.Web/Views/Admin/Groups/GroupsGrid.cshtml (hunk 2 above)
Comment (AJIex1s) [reply SchoolJournal.Web/Views/Admin/Groups/GroupsGrid.cshtml:189]: done?

File: SchoolJournal.Web/wwwroot/css/groups.css (hunk 3 above)
Comment (AJIex1s) [reply SchoolJournal.Web/wwwroot/css/groups.css:74]: without content title not working?

Comment (Mallusca) [reply SchoolJournal.Web/wwwroot/css/groups.css:74]: yes)

File: SchoolJournalCore.Domain/GroupDomainModel.cs (hunk 7 above)
Comment (Mallusca) [reply SchoolJournalCore.Domain/GroupDomainModel.cs:20]: done

File: SchoolJournalCore.ViewModels/Groups/GroupViewModel.cs (hunk 8 above)
Comment (Mallusca) [reply SchoolJournalCore.ViewModels/Groups/GroupViewModel.cs:20]: done

File: SchoolJournal.Web/Views/Admin/Groups/GroupsGrid.cshtml [hunk 9]
```diff
@@ -109,7 +111,7 @@
 
     @foreach (var group in Model.Groups)
     {
-        <tr id="group-@group.Id">
+        <tr id="group-@group.Id" student-in-group="@group.ContainsStudents.ToString()">
```
Comment (AJIex1s) [inline SchoolJournal.Web/Views/Admin/Groups/GroupsGrid.cshtml:114]: student-in-group="@group.ContainsStudents.ToString() - meaning\purpose?

File: SchoolJournal.Web/wwwroot/css/groups.css [hunk 10]
```diff
@@ -65,3 +65,17 @@ textarea {
...
     width: 130px;
 }
+
+.disabled {
+    color: grey;
+}
+    .disabled:hover {
+        color: grey;
+        content: attr(title);
+        pointer-events: none;
```
Comment (AJIex1s) [inline SchoolJournal.Web/wwwroot/css/groups.css:75]: pointer-events: none; why you disable pointer events on hover?

File: SchoolJournalCore.DAL.Interfaces/IGroupsRepository.cs [hunk 11]
```diff
@@ -13,9 +13,6 @@ public interface IGroupsRepository
     {
         IQueryable<GroupDomainModel> GetGroups();
         Task<long> CreateGroup(CreateGroupDomainModel model);
-        public IQueryable<string> GroupNames();
```
Comment (AJIex1s) [inline SchoolJournalCore.DAL.Interfaces/IGroupsRepository.cs:16]: why did you delete this?

File: SchoolJournalCore.DAL/Models/Group.cs [hunk 12]
```diff
@@ -21,5 +21,6 @@ public class Group
         public virtual Teacher Teacher { get; set; }
 
         public virtual ICollection<WorkingHours> WorkingHours { get; set; }
+        public virtual ICollection<Student> Students { get; set; }
```
Comment (AJIex1s) [inline SchoolJournalCore.DAL/Models/Group.cs:24]: formatting - add line break

File: SchoolJournal.Web/wwwroot/css/groups.css (hunk 10 above)
Comment (AJIex1s) [reply SchoolJournal.Web/wwwroot/css/groups.css:75]: color: grey; why are you duplicating this?

File: SchoolJournalCore.DAL/Models/Group.cs (hunk 12 above)
Comment (Mallusca) [reply SchoolJournalCore.DAL/Models/Group.cs:24]: done

File: SchoolJournalCore.DAL.Interfaces/IGroupsRepository.cs (hunk 11 above)
Comment (Mallusca) [reply SchoolJournalCore.DAL.Interfaces/IGroupsRepository.cs:16]: recovered

File: SchoolJournal.Web/Views/Admin/Groups/GroupsGrid.cshtml (hunk 9 above)
Comment (Mallusca) [reply SchoolJournal.Web/Views/Admin/Groups/GroupsGrid.cshtml:114]: done, delete

File: SchoolJournal.Web/wwwroot/css/groups.css (hunk 10 above)
Comment (Mallusca) [reply SchoolJournal.Web/wwwroot/css/groups.css]: done

File: SchoolJournal.Web/wwwroot/css/groups.css [hunk 13]
```diff
@@ -65,3 +65,17 @@ textarea {
     height: 32px;
     width: 130px;
 }
+
+.disabled {
+    color: grey;
+    pointer-events: none;
+}
+    /*.disabled:hover {
```
Comment (AJIex1s) [inline SchoolJournal.Web/wwwroot/css/groups.css:73]: comment

File: SchoolJournalCore.DAL.Interfaces/IGroupsRepository.cs (hunk 11 above)
Comment (AJIex1s) [reply SchoolJournalCore.DAL.Interfaces/IGroupsRepository.cs:16]: used?

Comment (AJIex1s) [reply SchoolJournalCore.DAL.Interfaces/IGroupsRepository.cs:16]: GetGroupNames

File: SchoolJournal.Web/wwwroot/css/groups.css [hunk 14]
```diff
@@ -65,3 +65,17 @@ textarea {
...
+        content: attr(title);
+        
+    }*/
+    .disabled.icon-hover {
+        -webkit-transform: scale(1);
+        -ms-transform: scale(1);
+        transform: scale(1);
```
Comment (AJIex1s) [inline SchoolJournal.Web/wwwroot/css/groups.css:80]: use default if exists

Comment (Mallusca) [reply SchoolJournal.Web/wwwroot/css/groups.css:80]: done

File: SchoolJournal.Web/wwwroot/css/groups.css (hunk 13 above)
Comment (Mallusca) [reply SchoolJournal.Web/wwwroot/css/groups.css:73]: done
